File: solutions/dart/resistor-color-duo/1/lib/resistor_color_duo.dart
class ResistorColorDuo {
  Map<String, int> colors = {
    "black": 0,
    "brown": 1,
    "red": 2,
    "orange": 3,
    "yellow": 4,
    "green": 5,
    "blue": 6,
    "violet": 7,
    "grey": 8,
    "white": 9,
  };

  int value(List<String> colors) {
    int firstColor = this.colors[colors[0]]!;
    int secondColor = this.colors[colors[1]]!;
    return int.parse('$firstColor$secondColor');
  }
}

File: solutions/python/resistor-color-duo/1/resistor_color_duo.py
"""
This module defines a function to calculate the value of a resistor based on its color code.
"""

colors_list=[
    'black', 'brown', 'red',
    'orange', 'yellow', 'green',
    'blue', 'violet', 'grey',
    'white'
]

def value(colors):
    """
    This function takes a list of colors and returns a value.
    """
    res=''

    value_1=str(colors_list.index(colors[0]))
    value_2=str(colors_list.index(colors[1]))

    res=int(value_1+value_2)
    return res
